theSrc/scripts/LabeledScatter.js

```diff
@@ -83,30 +83,78 @@ class LabeledScatter {
-      const plot_config = { displayModeBar: false, editable: false }
-
-      let plotlyChart = await Plotly.react(this.rootElement, plot_data, plot_layout, plot_config)
-      const tmp_layout = {}
-      const is_legend_elements_to_right_of_plotly_legend = this.isLegendElementsToRightOfPlotlyLegend(plotlyChart._fullLayout, config)
-      if (is_legend_elements_to_right_of_plotly_legend && config.marginAutoexpand) {
-        const nsewdrag_rect = this.nsewdragRect()
-        const legend_right = config.colorScale !== null ? this.plotlyColorbarRect().right : this.plotlyLegendRect().right
-        const required_margin = (legend_right - nsewdrag_rect.right) + legend_points_and_bubble_legend_width
-        tmp_layout['margin.r'] = Math.max(required_margin, config.marginRight)
-      }
-      if (Object.keys(tmp_layout).length > 0) plotlyChart = await Plotly.relayout(plotlyChart, tmp_layout)
-      await this.drawScatterLabelLayer(plotlyChart._fullLayout, plotlyChart._fullData, config, is_legend_elements_to_right_of_plotly_legend)
-
-      if (FitLine.isFitDataAvailable(config)) {
-        FitLine.draw(this.rootElement, config)
+      const plot_config = {
+        displayModeBar: false,
+        editable: false,
+        edits: { annotationTail: true, legendPosition: false }
       }
 
-      plotlyChart.on('plotly_afterplot', () => {
-        this.drawScatterLabelLayer(plotlyChart._fullLayout, plotlyChart._fullData, config, is_legend_elements_to_right_of_plotly_legend)
-      })
-
-      this.addMarkerClickHandler()
-    } catch (err) {
+      let plotlyChart = await Plotly.react(this.rootElement, plot_data, plot_layout, plot_config)
+      if (Array.isArray(config.panels)) {
+        await this.drawSmallMultipleLabels(plotlyChart, config)
+
+        // Event handler for legendtoggle
+        let lastevent = ''
+        plotlyChart.on('plotly_legendclick', async (data) => {
+          lastevent = 'legendclick'
+          const annotations = plotlyChart._fullLayout.annotations
+          await Plotly.relayout(plotlyChart, { annotations: this.applyLegendClick(annotations, data, config) })
+          lastevent = 'legendclick'
+        })
+
+        // Event handler for Reset button
+        plotlyChart.on('plotly_clickannotation', async (data) => {
+          console.log(data)
+          if (data.annotation.text === 'Reset') {
+            this.stateObj.resetState()
+            await this.drawSmallMultipleLabels(plotlyChart, config)
+            lastevent = 'clickreset'
+          }
+        })
+
+        // Event handler for dragging and toggling scatter labels
+        // But do not save legend toggle
+        plotlyChart.on('plotly_afterplot', () => {
+          console.log('last event: ' + lastevent)
+          if (lastevent === '') {
+            this.stateObj.saveToState({ 'plotlyAnnotations': plotlyChart._fullLayout.annotations
+            .filter(a => a.showarrow && !(a.ax === 0 && a.ay === 0 && a.visible))
```

Comment:
Is this predicate correct? Should it be `!a.showarrow && a.visible`? Why do we check that `ax` and `ay` are zero?

Reply:
`a.showarrow` is true for the scatter marker labels - its used to distinguish them from the panel and axes titles. ax = 0, ay = 0  and  a.visible is not saved because these are the default state, i.e. the user hasn't changed them (if the user clicks on the marker then the labels can become hidden which we do save)

Reply:
I remove the scatter marker labels in the default state (ax = 0 & ay = 0 & visible = true) because it saves space, and also because I need to be able to check whether the user has made any changes - to detemine whether or not to show the reset button

Reply:
OK. We might need other ways to distinguish the annotations, when we start using annotations for other purposes like the subtitle and footer.